src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -106,17 +112,35 @@ LinalgBackendBase* infer_backend(const Container<T>& a, const Container<T>& b)
  * and a shallow-copy of SGVector with vector on CPU if GPU backend not available
  */
 template <typename T>
-SGVector<T> to_gpu(const SGVector<T>& vector)
+void to_gpu(SGVector<T>& a, SGVector<T>& b)
 {
```

Comment:
yeah so think about to_gpu being totally not threadsafe... that is something that you dont want to have on user code side... what you will mutex it constantly around the checking for on_gpu?